orchestrator/lambdas/layers/heimdall_utils/heimdall_utils/utils.py

```diff
@@ -96,18 +100,25 @@ def parse_timestamp(timestamp: Optional[str] = None) -> str:
     Notes:
         - The function uses the current system time to calculate the 3-month offset.
         - All returned timestamps are in UTC (denoted by the 'Z' suffix).
-        - The function logs a warning message when generating a default timestamp.
     """
-    format = "%Y-%m-%dT%H:%M:%SZ"
-    try:
-        if timestamp and bool(datetime.strptime(timestamp, format)):
-            return timestamp
-    except (TypeError, ValueError):
-        log.error("Timestamp is invalid")
-
-    log.warning("Generating Default timestamp")
-    current_timestamp = datetime.now()
-    three_months_ago = current_timestamp - timedelta(days=90)
-    result = three_months_ago.timestamp()
+    if timestamp and is_valid_timestamp(timestamp):
+        return timestamp
+
+    log.debug("Generating Default timestamp")
+    default_timestamp = get_datetime_from_past(90)
+    result = default_timestamp.timestamp()
+
+    return time.strftime(TIMESTAMP_FORMAT, time.gmtime(result))
 
-    return time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime(result))
+
+def is_valid_timestamp(timestamp: str) -> bool:
+    default_timestamp = get_datetime_from_past(90)
```

Comment:
Leaving this at exactly 90 days might leave room for a race condition.
- Heimdall runs weekly, and it sees a scan for a given repo that expires 1 hour from now (so it doesn't trigger a rescan)
- An hour later, downstream services purge data due to being outside the 90 day retention period
- Until the next scan a week later, we do not have data in downstream service for the given repo

To solve this, we could set this to 83 days. Let's make sure to document why though, as 83 would definitely seem like a magic number.